### DISCUSSION TITLE
Question about deleting a field in a nested object in Firebase Firestore in the extension task

### DISCUSSION THREAD

**Discussion by benmliang** [AntlerVC/firetable#1110] [Support / Q&A] [answered]
I am currently using rowy for my next project and I have a quick question to ask about deleting a field in a nested object in Firebase Firestore.

I have been using the `deleteField()` method and the `firebase.firestore.FieldValue.delete()` method in the extension body, but it seems that these methods do not work. Is there a way to delete a field in a nested object using these methods?

My current solution is to set the field to null using the Firestore update method, but this effectively "deletes" the field by removing its value, but the field itself will still exist in the Firestore document.

Please let me know if there is a way to delete a field in a nested object using the deleteField() or firebase.firestore.FieldValue.delete() method, or if there is an alternative method that I can use.

Thank you for your help,
Ben Liang

my code example:
```
const extensionBody: TaskBody = async({row, db, change, triggerType, ref}) => {
  if (triggerType === "delete") {
    await ref.parent.parent.update({
      [`noteSummary.${ref.id}`]: null
    })
  } else {
    await ref.parent.parent.update({
      [`noteSummary.${ref.id}`]: row
    })
  }
}
```

**Comment by BeeBombshell** [comment, marked as answer]
Hey @benmliang, to use `firebase.firestore` methods, we'd first have to manually import the Firestore library:
```js
import("@google-cloud/firestore")
```

For more insight on this, check out the example in this discussion: https://github.com/rowyio/rowy/discussions/1102
